src/averageAges.js

```diff
@@ -54,11 +75,23 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  let children = people
+    .filter(person => people
+      .find((motherName) => motherName.name === person.mother));
+
+  if (onlyWithSon) {
+    children = children.filter(person => person.sex === MAN_GENDER);
+  }
+
+  return Math.round(children.reduce((sum, person) => {
+    const mother = people
+      .find((motherName) => motherName.name === person.mother);
```

Comment:
```suggestion
      .find((mother) => mother.name === person.mother);
```
motherName.name looks strange, just mother